lib/Fmt_ast.ml

```diff
@@ -541,6 +541,10 @@ let sequence_blank_line c (l1 : Location.t) (l2 : Location.t) =
       loop l1.loc_end (Cmts.remaining_before c.cmts l2)
   | `Compact -> false
 
+(* Breaking before an attribute can confuse ocp-indent that will produce a
+   suboptimal indentation. *)
+let pre_attributes c = fmt_or c.conf.ocp_indent_compat " " "@ "
```

Comment:
I'm sorry for not thinking of this suggestion before, but I am inclined to try a different implementation, where instead of adding the `pre_attributes` function, `fmt_attributes` is changed to test if `pre` is "@ " and if so change it to " ". That will more tightly localize the mess made by ocp-indent compatibility.

Reply:
Good point, it looks better indeed. No regression with test_branch.